PacketDotNet/LSA.cs

```diff
@@ -0,0 +1,1165 @@
+    /// <summary>
+    /// AS-external-LSAs are the Type 5 LSAs.  These LSAs are originated by
+    /// AS boundary routers, and describe destinations external to the AS.
+    /// </summary>
+    public class ASExternalLSA : LSA
+    {
+        public static readonly LSAType lsaType = LSAType.ASExternal;
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public ASExternalLSA()
+        {
+            byte[] b = new byte[ASExternalLSAFields.MetricPosition];
+            this.header = new ByteArraySegment(b);
+            this.LSType = lsaType;
+            this.Length = (ushort)this.header.Bytes.Length;
+        }
+
+        /// <summary>
+        /// Constructs an ASExternalLSA with a list of ASExternalLinks
+        /// </summary>
+        public ASExternalLSA(List<ASExternalLink> links)
+        {
+            int length = ASExternalLSAFields.MetricPosition + ASExternalLink.Length * links.Count;
+            int offset = ASExternalLSAFields.MetricPosition;
+            byte[] b = new byte[length];
+
+            foreach (ASExternalLink l in links)
+            {
+                Array.Copy(l.Bytes, 0, b, offset, ASExternalLink.Length);
+                offset += ASExternalLink.Length;
+            }
+
+            this.header = new ByteArraySegment(b);
+            this.LSType = lsaType;
+            this.Length = (ushort)this.header.Bytes.Length;
+        }
+
+        /// <summary>
+        /// Constructs a packet from bytes and offset and length
+        /// </summary>
+        /// <param name="Bytes">
+        /// A <see cref="System.Byte"/>
+        /// </param>
+        /// <param name="Offset">
+        /// A <see cref="System.Int32"/>
+        /// </param>
+        /// <param name="length">
+        /// A <see cref="System.Int32"/>
+        /// </param>
+        public ASExternalLSA(byte[] packet, int offset, int length) :
+            base(packet, offset, length)
+        {
+
+        }
+
+        /// <summary>
+        /// The IP address mask for the advertised destination.
+        /// E.g. class A network - 255.0.0.0
+        /// </summary>
+        public IPAddress NetworkMask
+        {
+            get
+            {
+                var val = EndianBitConverter.Little.ToUInt32(header.Bytes, header.Offset + ASExternalLSAFields.NetworkMaskPosition);
+                return new System.Net.IPAddress(val);
+            }
+            set
+            {
+                byte[] address = value.GetAddressBytes();
+                Array.Copy(address, 0,
+                           header.Bytes, header.Offset + ASExternalLSAFields.NetworkMaskPosition,
+                           address.Length);
+            }
+        }
+
+        /// <summary>
+        /// The list of the AS-External-Links contained in this OSPF packet.
+        /// </summary>
+        public List<ASExternalLink> ASExternalLinks
+        {
+            get
+            {
+                int linkCnt = (this.Length - 4 - 20) / 12;

```

Comment:
it would be good to replace these numerical constants with class constants so the names of the class constants could help clarify why you are performing the adjustment of length